### PR TITLE
Normalize org name and external id before creating organization

### PR DESCRIPTION
**Pull request by bmaden-usds**
## Related Issue or Background Info

Fixes https://github.com/CDCgov/prime-simplereport/issues/2564

## Changes Proposed

- Normalize org name and external id before creating organization

## Additional Information

- This is just part of a continuing process to improve our detection and handling of duplicate organizations.

### UI
- [x] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 1]
```diff
@@ -148,9 +155,20 @@ private String checkForDuplicateOrg(String organizationName, String state, Strin
...
-        state, organizationName.replace(' ', '-').replace(':', '-'), UUID.randomUUID().toString());
+    organizationName =
+        organizationName
+            // remove all non-alpha-numeric
+            .replaceAll("[^-A-Za-z0-9 ]", "")
+            // spaces to hyphens
+            .replace(' ', '-')
+            // reduce repeated hyphens to one
+            .replaceAll("-+", "-")
+            // remove leading hyphens
```
Comment (nickclyde) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:166]: Thanks for the comments :100:

File: backend/src/test/java/gov/cdc/usds/simplereport/api/AccountRequestControllerTest.java [hunk 2]
```diff
@@ -153,6 +156,79 @@ void accountRequestSuccessful() throws Exception {
     assertThat(externalIdCaptor.getValue()).contains("RI-Day-Hayes-Trading-");
   }
 
+  @Test
+  void submitOrganizationAccountRequest_nameCleaning_success() throws Exception {
+    String requestBody =
+        createAccountRequest(
+            " Central   Schools  ",
```
Comment (nickclyde) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/AccountRequestControllerTest.java:163]: :laughing: I wouldn't put it past our users

**Review by nickclyde — APPROVED** [review]
Looks great!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 3]
```diff
@@ -98,7 +98,7 @@ public AccountResponse submitOrganizationAccountRequest(
             "This email address is already associated with a SimpleReport user.");
       } else {
         throw new BadRequestException(
-            "An unknown error occured when creating this organization in Okta.");
+            "An unknown error occurred when creating this organization in Okta.");
```
Comment (emmastephenson) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:101]: I think you'll also need to update [utils.tsx](https://github.com/CDCgov/prime-simplereport/blob/5c7eddb272de31c3cbbf4c839b4ec09cfce1705e/frontend/src/app/signUp/Organization/utils.tsx) on the frontend to make sure this message is being parsed correctly.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java [hunk 4]
```diff
@@ -148,9 +155,20 @@ private String checkForDuplicateOrg(String organizationName, String state, Strin
...
+            // remove all non-alpha-numeric
+            .replaceAll("[^-A-Za-z0-9 ]", "")
+            // spaces to hyphens
+            .replace(' ', '-')
+            // reduce repeated hyphens to one
+            .replaceAll("-+", "-")
+            // remove leading hyphens
+            .replaceAll("^-+", "");
+    if (organizationName.length() == 0) {
+      throw new BadRequestException("The organization name is invalid.");
```
Comment (emmastephenson) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:169]: We might also want to add a frontend toast for this message.

**Review by emmastephenson — CHANGES_REQUESTED** [review]
Overall LGTM, thanks so much! Just some minor comments about updating utils.tsx on the frontend so that these new BadRequestException messages get passed along appropriately, otherwise looks great.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java (hunk 3 above)
Comment (bmaden-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java:101]: Oh, good catch!  I didn't realize these messages were coupled to the frontend.

**Review by emmastephenson — APPROVED** [review]

File: frontend/src/app/signUp/Organization/utils.tsx [hunk 5]
```diff
@@ -147,6 +147,15 @@ export const organizationSchema: yup.SchemaOf<OrganizationCreateRequest> = yup
 
 export const organizationBackendErrors = (error: string): ReactElement => {
   switch (error) {
+    // Unusable org name.  Name probably doesn't contain any alphanumeric characters
+    case "The organization name is empty.":
+    case "The organization name is invalid.":
+      return (
+        <Alert type="error" title="Invalid organization name">
+          The organization name you entered is invalid. Please double check and
+          re-enter the organization name.
```
Comment (emmastephenson) [inline frontend/src/app/signUp/Organization/utils.tsx:156]: @Rebecca-LM Does this error message look alright to you? It's for the toast users will see if they enter an invalid name (one that's empty or only consists of special characters, for example)

**Comment by emmastephenson** [comment]
LGTM! I asked Rebecca to take a pass at the new error message but it's totally fine if you want to merge this first and update the message later; this is important enough that I'd rather get it in ASAP.

File: frontend/src/app/signUp/Organization/utils.tsx (hunk 5 above)
Comment (Rebecca-LM) [reply frontend/src/app/signUp/Organization/utils.tsx:156]: Yes LGTM! I sent this copy to Brett :)

**Pull request merged**